src/main/java/org/influxdb/impl/InfluxDBImpl.java

```diff
@@ -810,6 +835,18 @@ private boolean checkMessagePackSupport() {
     return (major >= 2) || ((major == 1) && (minor >= fromMinor));
   }
 
+  private boolean checkFormEncodedQuerySupport() {
+    Matcher matcher = APPLICATION_VERSION_PATTERN.matcher(version());
+    if (!matcher.find()) {
+      return false;
+    }
+    String[] versionNumbers = matcher.group().split("\\.");
+    final int major = Integer.parseInt(versionNumbers[0]);
+    final int minor = Integer.parseInt(versionNumbers[1]);
+    final int fromMinor = 3;
+    return (major >= 2) || ((major == 1) && (minor >= fromMinor));
+  }
```

Comment:
The current version of InfluxDB 2 OSS doesn't return correct version. See:

```console
docker run --rm --publish 8086:8086 influxdb:2.0-alpine

curl -i -X GET http://localhost:8086/ping

> HTTP/1.1 204 No Content
> X-Influxdb-Build: cloud2
> X-Influxdb-Version: 
> Date: Tue, 31 Aug 2021 06:49:18 GMT
```

What do you think about changing the `checkFormEncodedQuerySupport` to optimistic variant? The encoding query into body will be supported for `unknown` version and all version greater or equal to `1.3`.

@rhajek, @majst01